src/Http/Headers/src/ContentDispositionHeaderValue.cs

```diff
@@ -444,7 +444,7 @@ private StringSegment Sanitize(StringSegment input)
                 for (int i = 0; i < result.Length; i++)
                 {
                     var c = result[i];
-                    if ((int)c > 0x7f)
+                    if ((int)c > 0x7f || c == '\n')
```

Comment:
Maybe
```suggestion
                    if ((int)c > 0x7f || (int)c < 0x20)
```
?

Reply:
I like the idea, couldn't find this in the header spec though.

Reply:
Filtering control characters makes sense to me. We have filename* for preserving the full name.